app/src/main/java/com/example/marvelapp/mvvm/viewmodel/CharacterListViewModel.kt

```diff
@@ -37,19 +37,26 @@ class CharacterListViewModel(private val model: CharacterListModel) : ViewModel(
         }
     }
 
+    fun onCharacterPressed(characterId: String) {
+        _characterState.value = CharacterData(CharacterState.CHARACTER_PRESSED, idCharacter = characterId)
+    }
+
     data class CharacterData(
-        var state: CharacterState,
-        var data: List<Character> = emptyList()
+        val state: CharacterState,
+        val data: List<Character> = emptyList(),
+        val idCharacter: String = EMPTY_STRING
     )
 
     enum class CharacterState {
         RESPONSE_LOADING,
         RESPONSE_SUCCESS,
         RESPONSE_ERROR,
-        RESPONSE_ERROR_NOT_FOUND
+        RESPONSE_ERROR_NOT_FOUND,
+        CHARACTER_PRESSED
     }
 
     companion object {
         private const val NOT_FOUND = "NOT_FOUND"
+        private const val EMPTY_STRING = ""
```

Comment:
you have these constants in several places. Isn't better to create a ConstantUtils object and have all the constants there?

Reply:
done